src/charm.py

```diff
@@ -572,11 +572,24 @@ def _reconcile_runners(self, runner_manager: RunnerManager) -> Dict[str, Any]:
             return {"delta": {"virtual-machines": 0}}
 
     @retry(tries=10, delay=15, max_delay=60, backoff=1.5, local_logger=logger)
-    def _install_deps(self) -> None:
-        """Install dependencies."""
+    def _install_deps(self, upgrade_compliance: bool = False) -> None:
+        """Install dependencies.
+
+        Args:
+            upgrade_compliance: Upgrade (True) or install (False)
+                the repo-policy-compliance package.
+
+        """
         logger.info("Installing charm dependencies.")
 
         # Binding for snap, apt, and lxd init commands are not available so subprocess.run used.
+        execute_command(["/usr/bin/apt-get", "update"])
+        # install dependencies used by repo-policy-compliance and the firewall
+        execute_command(
+            ["/usr/bin/apt-get", "install", "-qy", "gunicorn", "python3-pip", "nftables"]
+        )
+
+        # Install repo-policy-compliance package.
```

Comment:
It's possible that relocating the installation process before configuring the proxy environment variables could cause some issues.

Reply:
Got it but isn't 
```
execute_command(
   ["/usr/bin/apt-get", "install", "-qy", "gunicorn", "python3-pip", "nftables"]
)
``` 
ignoring/not using the `env` though?

I think only having it under the `env` definition and not adding the `env` as a kwarg to the command does not allow the command to access the `env`. Right?

Reply:
I think the `execute_command` uses `subprocess.run` internally, and `subprocess.run` by default inherits the parent process's environment variables. So setting the `http_proxy` environment in the charm process (parent process) will affect the environment variables of the child process (`apt-get`).

Reply:
I also think that apt-get doesn't require the proxy

Reply:
@devspyrosv found out that the code didn't set the environment variables of the charm process, instead, it just create an `env` variable and passed to the `execute_command` function. So my statement is wrong in the last comment. 

I think the `env` variable in missing in the `execute_command(["/usr/bin/apt-get", "install", "-qy", "gunicorn", "python3-pip", "nftables"])`, since `apt-get` requires http proxies as well. Would you please add that, thanks!

Reply:
> I also think that apt-get doesn't require the proxy

It doesn't require proxy in our production and staging environment, but it does in a general environment.

Reply:
Preparing commit that passes env to all the `apt` and `pip` commands in that method. Wouldn't `snap` commands below them also need this?

Reply:
Note that Juju has specific model config for APT proxies:
https://juju.is/docs/juju/offline-mode-strategies#heading--apt-proxy
https://juju.is/docs/juju/configure-juju-for-offline-usage#heading--offline-configuration-methods

Reply:
@gregory-schiano Do you mean that environment vars like `http_proxy` and `no_proxy` should not be used inside the charm code and that proxying should be handled at the model level? Could you please elaborate a bit on the above?
Thx!

Reply:
I also think that `apt` will not honor the `http[s]_proxy` vars. Doesn't it need a modification in an etc conf file?

So I think that proxying in general needs rework and a discussion before it's fixed/improved. @jdkandersson should it be part of this PR?

Reply:
The Juju model can be configured to make units use proxies for apt and snap.

I believe we can consider this discussion complete.